### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by natLF3RD**
Frontend static design merge into master.
Unintegrated with backend.

Contains the following page designs:
- Building list page
- Room list page
- Room occupancy page

### CODE REVIEW

File: frontend/.browserslistrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+> 1%
+last 2 versions
+not dead
```
Comment (Choco3-3) [inline frontend/.browserslistrc:3]: What is the point of this file?

File: frontend/.gitignore [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+yarn-error.log*
+
+# Editor directories and files
+.idea
+.vscode
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
```
Comment (Choco3-3) [inline frontend/.gitignore:21]: Why is there a gitignore inside a folder when there is already a gitignore on the top level?

File: frontend/src/models/bindingModel.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Location: string;
+    RoomName: string;
+    // list of booked time slot where the room is not avaliable
+    BookedTimeSlots: Array<BookedSlots>;
+
+    constructor(json: any) {
+        this.Location = json.Location;
+        this.RoomName = json.RoomName;
+        this.BookedTimeSlots = [];
+        for(let room of json.BookedTimeSlots) {
```
Comment (Choco3-3) [inline frontend/src/models/bindingModel.ts:12]: The `for` loop style for this project includes a space after the `for` keyword

**Review by Choco3-3 — CHANGES_REQUESTED** [review]
Please refer to comments for suggestions and request for reasoning on specific changes.

File: frontend/.browserslistrc (hunk 1 above)
Comment (natLF3RD) [reply frontend/.browserslistrc:3]: Its  a file generated by the Vue cli. Used to indicate browser compatibility it seems.

File: frontend/src/models/bindingModel.ts (hunk 3 above)
Comment (natLF3RD) [reply frontend/src/models/bindingModel.ts:12]: fixed

File: frontend/.gitignore (hunk 2 above)
Comment (natLF3RD) [reply frontend/.gitignore]: I've merged it's contents into root now.

**Review by Choco3-3 — APPROVED** [review]
LGTM!

**Pull request merged**